### PR TITLE
Improve Tests and Documentation

### PR DESCRIPTION
**Pull request by Lurk390**
Fixes issues #48, #49, #50, #51 

Improved testing by simplifying files and using pyfixtures for performance. Also improved documentation:

1. Testing
    - Combined `test_minecraft.py` into `test_main.py`.
    - Added pytest fixtures at the class scope to remove redundant code and improve testing performance.

2. Documentation
    - Standardized docstrings format by removing unnecessary new lines and spaces and adding parameters and returns when appropriate.
    - Added class docstrings.
    - Included type hints for every function to improve code readability.

### CODE REVIEW

**Review by xieWright — CHANGES_REQUESTED** [review]
Currently, not too sure if this is an improvement to the current testing method. Would like to see a demonstration and have a vote on it Wed.

File: tempus_fugit_minecraft/main.py
Comment (w1158mxs) [inline tempus_fugit_minecraft/main.py:1]: what is the goal of specifying variables x,y,z, and n as float type? is that better for memory?

File: tests/test_main.py
Comment (w1158mxs) [inline tests/test_main.py:1]: you need to import dark_cloud also

File: tests/test_minecraft.py
Comment (w1158mxs) [inline tests/test_minecraft.py:1]: I see you merge tests in 1 file, Good one!

**Review by w1158mxs — APPROVED** [review]
good Job! Just import dark_clouds also

File: tests/test_main.py [hunk 1]
```diff
@@ -3,16 +3,90 @@
...
 @pytest.fixture(scope="class")
 def window():
-    window = Window()
-    return window
+    yield Window()
+
+
+@pytest.fixture(scope="class")
+def model():
+    yield Model()
```
Comment (bakursait2) [inline tests/test_main.py:16]: this actually very helpful as we won't bother creating instances of Window and Model classes on each method.
Good job on this.

File: tests/test_main.py
Comment (Lurk390) [reply tests/test_main.py:1]: This depends on who merges first, and the other person will have to fix the merge conflicts.

File: tempus_fugit_minecraft/main.py
Comment (Lurk390) [reply tempus_fugit_minecraft/main.py:1]: It needs to be a float to support decimal numbers.

**Review by xieWright — APPROVED** [review]
Alright, I understand what the changes do now. Improves testing to only have to use one model and window instead of having to create multiples. It has also passed the auto test, so it's good to go.

**Review by bakursait2 — APPROVED** [review]
Those changes are good and helpful to remove the redundancy of creating instances from `Window` and `Model` classes.

**Review by criticalglitch — APPROVED** [review]

**Pull request merged**